### PR TITLE
Add server integration tests

### PR DESCRIPTION
**Pull request by Prytu**
solves #120

### CODE REVIEW

**Comment by VirrageS** [comment]
You should add database to travis because it fails when trying to connect to database.
https://docs.travis-ci.com/user/database-setup/

**Comment by Prytu** [comment]
ye, I will in a moment, thanks!

File: .travis.yml [hunk 1]
```diff
@@ -4,7 +4,13 @@ go:
   - 1.7.3
 os:
   - linux
+
+services:
+  - postgresql
+
 before_script:
+  - psql -c 'create database travis_ci_test;' -U postgres -f postgres/test/schema.sql
```
Comment (VirrageS) [inline .travis.yml]: What is the point of using different schema for tests?

File: README.md [hunk 2]
```diff
@@ -77,3 +77,12 @@ Then, depending on the services we want to start we have to type:
...
 
     $ docker-compose -f docker/production.yml build && docker-compose -f docker/production.yml up
+    
+### Testing
+
+    $ docker-compose -f docker/test.yml build && docker-compose -f docker/test.yml up
+    
+    To clean up volumes of test database run:
+    
+    $ docker-compose -f test.yml down -v
```
Comment (VirrageS) [inline README.md]: It is a bit inconsistent with line above. There should be `-f docker/test.yml`

**Comment by Prytu** [comment]
fixed

**Pull request merged**